website/content/specification/syntax/constraints.md

```diff
@@ -12,6 +12,49 @@ TODO: P3: Address issue https://github.com/usnistgov/metaschema/issues/325
 
 ## Common Constraint Data
 
+## Let expressions
+
+Using the `let` element, a variable can be defined, which can be used in a Metapath expression in subsequent constraints.
+
+A `let` statement has a REQUIRED `@var` attribute, which defines the variable name.
+
+A `let` statement has a REQUIRED `@expression` attributes, which defines an Metapath expression, whose result is used to define the variable's value in the evaluation context.
+
+During constraint evaluation, each `let` statement MUST be evaluated in encounter order. If a previous variable is bound with the same name in the evaluation context, the new value MUST bound in a sub-context to avoid side effects. This sub-context MUST be made available to any constraints following the `let` statement declaration, and to any constraints defined on child nodes of the current context.
```

Comment:
Question: does this mean the parent `<constraint/>` is the root context for any `let` declaration?

Reply:
No. Constraints are evaluated against content nodes. Each content node corresponds to a Metaschema module definition, which has constraints. Each of these constraints are evaluated with the evaluation context's focus on that node. i.e., a constraints target is evaluated using that node as the focus.

How do I clarify that here? I do think we need a section talking about evaluation. Maybe there?

Reply:
OK, I need to think about this for a bit. I can tell framed that question very poorly. I guess I am interested in the definition of the `let`s and precedence of variables. I understand what you wrote but I must have phrased my confusion poorly. So, let me think about this before I pretend I can clarify it. 😆